### ISSUE TITLE
Switching between domains with SAML user

### ISSUE THREAD

**Issue by adrianopaterlini** [apache/cloudstack#5618]
##### ISSUE TYPE
 *  Bug Report

##### COMPONENT NAME

~~~
UI, SAML
~~~

##### CLOUDSTACK VERSION

~~~
4.16.0.0-SNAPSHOT
~~~

##### CONFIGURATION
SAML is enabled

##### OS / ENVIRONMENT
Using online environment http://primate-qa.cloudstack.cloud:8080/client/

##### SUMMARY
When a SAML user exists in more than one account, it is possible to switch from one domain to another, but the lateral menu is loaded based only on the first account permissions. (Per instance User Role don’t have Domain menu)

##### STEPS TO REPRODUCE
1. Add a SAML user to an Account with User Role
2. Add same SAML user to an Account with Domain Admin Role
3. Login with SAML user
4. Switch Domain

Using primate-qa environment the user Rick was created first into an account with a user role, and user Sheldon the first creation was into a domain admin account.

##### EXPECTED RESULTS

~~~
Switching between domains reload lateral menu.
~~~

##### ACTUAL RESULTS

~~~
Menu is loaded based only on the first account permissions.
~~~

**Comment by DaanHoogland** [comment]
@adrianopaterlini, sorry to not come back sooner. I am now looking into this. You said you tested this in the online env, but I see both rick and sheldon as users in the same account. 
I'm afraid I don't understand your user scenario.
- When you said switch domain, did you log of and log into andother domain?
- did the user move to the other account?

**Comment by adrianopaterlini** [comment]
@DaanHoogland 

In the case of SAML users, same user can belong to multiple accounts/domains, a combo box is loaded at the top to switch between domains, you don't need to log off.

![sheldon1](https://user-images.githubusercontent.com/15329040/148996418-fa785199-4d3a-4d11-b774-071196642307.png)

For example, if you login using Sheldon (Single Sign On tab), switch to rick account (saml domain), "domain" is still shown at the left bar menu, but since at this domain the user dont have permissions, we got the screen:
 
![sheldon2](https://user-images.githubusercontent.com/15329040/148997403-6d9a1051-9df9-432e-a8a3-5e41bb5e77f1.png)

In the other hand, when using Rick, even after switching to a domain with higher permissions, left menu is not reloaded accordingly.

> @adrianopaterlini, sorry to not come back sooner. I am now looking into this. You said you tested this in the online env, but I see both rick and sheldon as users in the same account. I'm afraid I don't understand your user scenario.
> 
> * When you said switch domain, did you log of and log into andother domain?
> * did the user move to the other account?

**Comment by DaanHoogland** [comment]
I see @adrianopaterlini so the action on the dropdown should do some more work, clear.

**Comment by utchoang** [comment]
@DaanHoogland I did the test on QA but didn't see the same error, do you have the same problem as above @shwstppr ?
https://user-images.githubusercontent.com/13766648/149120981-2bae518e-c991-4e9d-af9e-434b00ada720.mp4

**Comment by DaanHoogland** [comment]
@utchoang the point made by @adrianopaterlini is that
1. if you log in as user A then
2.  switch to user B, B having less rights,
3.  you still see the same items in the menu but accessing them gives errors
however
1. if you log in as user B
2. you don't see the menu item which you are allowed to (fine) but
3. if you switch to user A that does have the extra right
4. you still don't see the items you now should have access too.

I think this is a valid concern.

**Comment by adrianopaterlini** [comment]
@utchoang 

Following 1-3 steps by @DaanHoogland 

https://user-images.githubusercontent.com/15329040/149207567-e459b4f8-5594-423f-b2ea-1713426df4ee.mp4

**Comment by utchoang** [comment]
@DaanHoogland cc @rohityadavcloud @shwstppr  I tried to check again, but I see the address returned after the successful login is changed. This makes it impossible for me to test successfully in my environment. Is there any way to change it?
![image](https://user-images.githubusercontent.com/13766648/149257980-b7f56573-3d6a-4cb8-9585-605fb086ead8.png)
![image](https://user-images.githubusercontent.com/13766648/149258014-1d000727-6d4d-40e4-910b-1344d3b414ae.png)
![image](https://user-images.githubusercontent.com/13766648/149258056-87e29536-f4ac-49e9-9229-7437d1d4d224.png)

**Comment by DaanHoogland** [comment]
@utchoang are you asking about the url/redirect? I think you would only need the user/account, would you?

**Comment by utchoang** [comment]
@DaanHoogland No, you can see my PR https://github.com/apache/cloudstack/pull/5855 to test the cases then I need the  redirect to the link http://qa.cloudstack.cloud:8080/client/pr/5855 after login successfully but now it is auto redirecting about http://primate-qa.cloudstack.cloud:8080. This caused my changes to not be applied.

**Comment by DaanHoogland** [comment]
Ah, i see now why. I'll try and build your PR locally later today. Hope it works like that.

**Comment by adrianopaterlini** [comment]
@utchoang @DaanHoogland 

I did some testing and LGTM. It is just slow, but that is something being addressed already (https://github.com/apache/cloudstack/issues/5266)

(I have bypassed saml redirections, by first login in using http://primate-qa.cloudstack.cloud:8080/client/, and just after it manually going to http://primate-qa.cloudstack.cloud:8080/client/pr/5855/)

![pr-5855](https://user-images.githubusercontent.com/15329040/149341771-1918f2dc-1329-48e9-9f86-3c46be8af6a1.png)

**Comment by DaanHoogland** [comment]
Good @adrianopaterlini thanks for testing this.

**Comment by sureshanaparti** [comment]
Fixed in #5855
